src/main/java/team4/footwithme/team/service/TeamServiceImpl.java

```diff
@@ -2,8 +2,114 @@
 
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import team4.footwithme.member.domain.Gender;
+import team4.footwithme.member.domain.Member;
+import team4.footwithme.member.repository.MemberRepository;
+import team4.footwithme.team.api.request.TeamAddResquest;
+import team4.footwithme.team.domain.TeamMember;
+import team4.footwithme.team.domain.TeamRate;
+import team4.footwithme.team.repository.TeamMemberRepository;
+import team4.footwithme.team.repository.TeamRateRepository;
+import team4.footwithme.team.service.response.TeamInfoResponse;
+import team4.footwithme.team.domain.Team;
+import team4.footwithme.team.domain.TotalRecord;
+import team4.footwithme.team.repository.TeamRepository;
+import team4.footwithme.team.service.response.TeamCreatedResponse;
+
+import java.util.ArrayList;
+import java.util.List;
 
 @RequiredArgsConstructor
 @Service
 public class TeamServiceImpl implements TeamService{
+
+    private final TeamRepository teamRepository;
+    private final TeamRateRepository teamRateRepository;
+    private final TeamMemberRepository teamMemberRepository;
+    private final MemberRepository memberRepository;
+
+    @Override
+    @Transactional
+    public TeamCreatedResponse createTeam(TeamAddResquest dto) {
+        /**
+         * # 1.
+         * chatRoomId는 일단 가짜 id 넣어놓고,
+         * 채팅방 생기면 해당 id 적용
+         */
+        Long stadiumId = null;
+        Long chatRoomId = 100000L;
+
+        //TotalRecord 초반 셋팅: count 0
+        TotalRecord totalRecord = TotalRecord.builder()
+                .winCount(0)
+                .drawCount(0)
+                .loseCount(0)
+                .build();
+
+        //dto -> entity
+        Team entity = Team.create(
+                stadiumId,
+                chatRoomId,
+                dto.name(),
+                dto.description(),
+                totalRecord.getWinCount(),
+                totalRecord.getDrawCount(),
+                totalRecord.getLoseCount(),
+                dto.location()
+        );
+        Team createdTeam = teamRepository.save(entity);
+
+        return new TeamCreatedResponse(
+                createdTeam.getTeamId(),
+                createdTeam.getStadiumId(),
+                createdTeam.getChatRoomId(),
+                createdTeam.getName(),
+                createdTeam.getDescription(),
+                createdTeam.getTotalRecord().getWinCount(),
+                createdTeam.getTotalRecord().getDrawCount(),
+                createdTeam.getTotalRecord().getLoseCount(),
+                createdTeam.getLocation()
+        );
```

Comment:
저는 이렇게 길어지는게 싫어서 Response에서 of를 이용해서 작성합니다.
```suggestion
        return new TeamCreatedResponse.of(createTeam);
```

이렇게 작성하고 of메서드에서 매핑해서 사용하는 편입니다.

Reply:
아 이런 꿀팁이..! 감사합니다..!!!